pkg/controller/external.go

```diff
@@ -162,25 +162,26 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot get connection details")
 	}
 
+	lateInitedParams, err := tr.LateInitialize(res.State.GetAttributes())
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, "cannot late initialize parameters")
+	}
+	lateInitialized := annotationsUpdated || lateInitedParams
+	resourceReady := tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available())
 	// We try to mark the resource ready before the spec late initialization
 	// and a call for "Plan" because those operations are costly, i.e. late-init
 	// causes a spec update which defers status update to the next reconcile and
 	// "Plan" takes a few seconds.
-	if !tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available()) {
+	if !resourceReady || lateInitialized {
 		tr.SetConditions(xpv1.Available())
```

Comment:
nit: I noticed that it was already the case before, but, confused to mark **the resource as ready when the resource is not ready** :) it would be great if you would mind clarifying this as well while updating the above comment.

Reply:
Yes, that also confused me and caught my interest while preparing this PR. But as you mentioned this is already the case in main branch. I added a comment making this assumption explicit and maybe we can follow the implications of this assumption up with a separate issue.